### ISSUE TITLE
[Bug]: dns_query go.d plugin waits for timeout before reporting if at least one server is not responding

### ISSUE THREAD

**Issue by aldem** [netdata/netdata#15660]
### Bug description

If at least one server from a list configured in `dns_query.conf` times out then the plugin waits for at least `update_interval + timeout` seconds before producing a report even for servers which are responding, which leads to gaps in charts and spoils measurements.

Example configuration:

```
update_every: 1
autodetection_retry: 0
priority: 70000

jobs:
 - name: Google
   domains: [google.com]
   servers: [8.8.8.8, 8.8.1.1]
```

`8.8.1.1` is not responding to DNS queries, default timeout is 2 seconds, and plugin outputs (I removed chart definitions):

```
### First output

21:45:24.704778 BEGIN 'dns_query_Google.server_8_8_8_8_record_A_query_status'
21:45:24.704782 SET 'success' = 1
21:45:24.704785 SET 'network_error' = 0
21:45:24.704788 SET 'dns_error' = 0
21:45:24.704792 END
21:45:24.704795
21:45:24.704798
21:45:24.704802 BEGIN 'dns_query_Google.server_8_8_8_8_record_A_query_time'
21:45:24.704805 SET 'query_time' = 27857311
21:45:24.704810 END
21:45:24.704814
21:45:24.704818
21:45:24.704822 BEGIN 'dns_query_Google.server_8_8_1_1_record_A_query_status'
21:45:24.704825 SET 'success' = 0
21:45:24.704829 SET 'network_error' = 1
21:45:24.704833 SET 'dns_error' = 0
21:45:24.704836 END
21:45:24.704840
21:45:24.704843
21:45:24.704847 BEGIN 'dns_query_Google.server_8_8_1_1_record_A_query_time'
21:45:24.704851 SET 'query_time' =
21:45:24.704854 END
21:45:24.704858
21:45:24.704862 BEGIN 'netdata.execution_time_of_dns_query_Google'
21:45:24.704866 SET 'time' = 2000
21:45:24.704870 END
21:45:24.704874
21:45:25.294879
21:45:26.295409
21:45:27.295347

### Next one and all subsequent are now delayed by 3 seconds (update_interval + timeout)

21:45:27.704869 BEGIN 'dns_query_Google.server_8_8_8_8_record_A_query_status' 3000082
21:45:27.704891 SET 'success' = 1
21:45:27.704898 SET 'network_error' = 0
21:45:27.704904 SET 'dns_error' = 0
21:45:27.704909 END
21:45:27.704914
21:45:27.704919 BEGIN 'dns_query_Google.server_8_8_8_8_record_A_query_time' 3000082
21:45:27.704925 SET 'query_time' = 28614992
21:45:27.704928 END
21:45:27.704933
21:45:27.704936 BEGIN 'dns_query_Google.server_8_8_1_1_record_A_query_status' 3000082
21:45:27.704940 SET 'success' = 0
21:45:27.704943 SET 'network_error' = 1
21:45:27.704946 SET 'dns_error' = 0
21:45:27.704950 END
21:45:27.704953
21:45:27.704956 BEGIN 'dns_query_Google.server_8_8_1_1_record_A_query_time'
21:45:27.704960 SET 'query_time' =
21:45:27.704964 END
21:45:27.704967
21:45:27.704971 BEGIN 'netdata.execution_time_of_dns_query_Google' 3000082
21:45:27.704975 SET 'time' = 2000
21:45:27.704979 END
...
```

Note that timing for responding server is reported properly, it is just delayed.

### Expected behavior

Timeouts must not delay reporting, I would expect that all queries for all jobs are done in parallel and once `timeout` is reached report is produced for all servers every `update_interval` seconds.

Probably it even makes sense to cap `timeout` to `update_interval` as it makes little sense to report every 1 second and have 2 seconds timeout, especially by default.

### Steps to reproduce

1. Configure dns_query plugin with one responding and one not responding servers (or use example above)
2. Run dns_query plugin
3. Observe the behavior described

### Installation method

kickstart.sh

### System info

```shell
System independent
```

### Netdata build info

```shell
v1.41.0
```

### Additional info

Actually even setting timeout to 1 second does not solve the problem - every report is delayed by 2 seconds instead of 3, and fractional timeouts are not supported.

Also it would be nice to be able to define global (default) timeout for all jobs, though this is a bit out of scope.

**Comment by ilyam8** [comment]
Hi, @aldem. This is by design and cannot be reported earlier if the servers are in the same data collection job and one of them has timed out.

> Probably it even makes sense to cap timeout to update_interval

Yes, we can do that.

> fractional timeouts are not supported

Supported. You can set it like "900ms".

**Comment by aldem** [comment]
Well, probably it is time to rethink the design then? Problem collecting one metric (dimension) should not affect others.

As to this specific case - I would suggest to change the default timeout to 500ms - this is well below default `update_interval` but at the same time well above typical response time for DNS servers.

Global setting for `timeout` would be nice as well, it is easy to forget to add it for a new job.

**Comment by ilyam8** [comment]
@aldem you need to split servers into multiple data collection jobs then. Even if we would report earlier we would have to delay the next data collection if timeout > update_every.

**Comment by aldem** [comment]
@ilyam8 What is the point to have more than than one server in one job then?

In my setup one job is a set of related servers, which is convenient to group/aggregate/monitor, if I split them then it would be less convenient to monitor specific groups, and I am pretty sure I am not alone in this.

**Comment by ilyam8** [comment]
Having more than one:
 - simplifies config.
 - all servers have the same `_collect_job` label.

I think apart from it there is no difference between single/multiple jobs. And what are you asking is exactly how multiple jobs work, if it is a single with multiple severs we need to wait until a timeout before the next data collection.

What I can do is ensure that timeout is always less than update_every (e.g. for 100ms) so there are no gaps in case of timeouts.

**Comment by ilyam8** [comment]
> it needs to be registered without affecting others

I understood you. I agree, but in this case, multiple jobs should be used, not one.
